components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/claims/impl/DefaultClaimHandler.java

```diff
@@ -580,13 +580,15 @@ protected Map<String, String> handleLocalClaims(String spStandardDialect,
         // Retrieve all non-null user claim values against local claim uris.
         allLocalClaims = retrieveAllNunNullUserClaimValues(authenticatedUser, claimManager, appConfig, userStore);
 
-        String applicationRoles = getApplicationRoles(authenticatedUser, context);
+        if (requestedClaimMappings.get(FrameworkConstants.APP_ROLES_CLAIM) != null) {
```

Comment:
also we need to handle the case of local role claim

Reply:
Fixed with 64b0ee85493b0d5e092942ed271af73cc7a6c463.

Reply:
Fixed with 2abab977bbda98ae43dc6d8631a86c848ded9e90